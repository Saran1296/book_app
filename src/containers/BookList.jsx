import React, { Component } from "react";

import Book from "../components/Book";
import { bookListConstant } from "../utils/constant";
export class BookList extends Component {
  state = {
    selectedBookList: null,
  };
  handleCellClick = (book) => {
    this.setState({ selectedBookList: book });
  };
  render() {
    const { selectedBookList } = this.state;
    return (
      <div>
        <h3>Welcome to my favourite book portal!</h3>
        <table className="booklist-table">
          <thead>
            <tr>
              <th>SI.No</th>
              <th>Book Name</th>
            </tr>
          </thead>
          <tbody>
            {bookListConstant.map((book) => {
              return (
                <tr key={book.id}>
                  <td>{book.id}</td>
                  <td
                    data-testid={`${book.id}-${book.name}`}
                    onClick={() => this.handleCellClick(book)}
                    style={{ cursor: "pointer", color: "blue" }}
                  >
                    {book.name}
                  </td>
                </tr>
              );
            })}
          </tbody>
        </table>
        <br />
        <hr />
        <Book bookDetails={selectedBookList} />
      </div>
    );
  }
}

export default BookList;
