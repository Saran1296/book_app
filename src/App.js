import './App.css';
import BookList from './containers/BookList';

function App() {
  return (
    <div className="App">
      <BookList/>
    </div>
  );
}

export default App;
