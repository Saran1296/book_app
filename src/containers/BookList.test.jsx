import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import BookList from "./BookList";
import App from "../App";

describe("BookList component", () => {
  it("to have welcome text- Welcome to my favourite book portal!", () => {
    render(<BookList />);
    screen.getByText("Welcome to my favourite book portal!");
  });
  it("should have the right snapshot", () => {
    const snapShotJson = renderer.create(<BookList />).toJSON();
    expect(snapShotJson).toMatchSnapshot();
  });
  it("render whole APP component to check the props working between both component", () => {
    render(<App />);
    const bookElement = screen.getByTestId("1-React JS Book");
    fireEvent.click(bookElement);
    expect(screen.getByText("Selected book details")).toBeInTheDocument();
    expect(screen.getByTestId("selected-bookName")).toHaveTextContent(
      "Selected Book Name: React JS Book"
    );
    expect(
      screen.getByText("Learn Class Components, Hooks")
    ).toBeInTheDocument();
  });
});
