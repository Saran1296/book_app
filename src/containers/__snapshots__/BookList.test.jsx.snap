// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList component it should have the right snapshot 1`] = `
<div>
  <h3>
    Welcome to my favourite book portal!
  </h3>
  <table
    className="booklist-table"
  >
    <thead>
      <tr>
        <th>
          SI.No
        </th>
        <th>
          Book Name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td
          data-testid="1-React JS Book"
          onClick={[Function]}
          style={
            Object {
              "color": "blue",
              "cursor": "pointer",
            }
          }
        >
          React JS Book
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td
          data-testid="2-Redux Book"
          onClick={[Function]}
          style={
            Object {
              "color": "blue",
              "cursor": "pointer",
            }
          }
        >
          Redux Book
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td
          data-testid="3-Jest + React testing library"
          onClick={[Function]}
          style={
            Object {
              "color": "blue",
              "cursor": "pointer",
            }
          }
        >
          Jest + React testing library
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <hr />
  <p>
    Click on any book name to explore more!
  </p>
</div>
`;

exports[`BookList component should have the right snapshot 1`] = `
<div>
  <h3>
    Welcome to my favourite book portal!
  </h3>
  <table
    className="booklist-table"
  >
    <thead>
      <tr>
        <th>
          SI.No
        </th>
        <th>
          Book Name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td
          data-testid="1-React JS Book"
          onClick={[Function]}
          style={
            Object {
              "color": "blue",
              "cursor": "pointer",
            }
          }
        >
          React JS Book
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td
          data-testid="2-Redux Book"
          onClick={[Function]}
          style={
            Object {
              "color": "blue",
              "cursor": "pointer",
            }
          }
        >
          Redux Book
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td
          data-testid="3-Jest + React testing library"
          onClick={[Function]}
          style={
            Object {
              "color": "blue",
              "cursor": "pointer",
            }
          }
        >
          Jest + React testing library
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <hr />
  <p>
    Click on any book name to explore more!
  </p>
</div>
`;
